src/digest.rs

```diff
@@ -239,6 +239,28 @@ pub struct Digest {
 }
 
 impl Digest {
+    /// Create a `Digest` from an algorithm and an already computed digest.
+    ///
+    /// The `data` needs to match the expected output length of the `algorithm`.
+    pub fn from_bytes(
+        algorithm: &'static Algorithm,
+        data: &[u8]
+    ) -> Result<Digest, crate::error::Unspecified> {
+        if algorithm.output_len != data.len() {
+            return Err(crate::error::Unspecified);
+        }
+
+        const SIZE: usize = 512 / 8 / core::mem::size_of::<BigEndian<u64>>();
```

Comment:
That's a bit dodgy and there is certainly a much better way. Do you have a recommendation?